Code/GraphMol/SmilesParse/CXSmilesOps.cpp

```diff
@@ -1300,6 +1300,19 @@ std::string quote_string(const std::string &txt) {
   return txt;
 }
 
+std::string quote_atomprop_string(const std::string &txt) {
+  // at a bare minimum, . needs to be escaped
+  std::string res;
+  for(auto c : txt) {
+    if(c == '.') {
+      res += "&#46;";
+    } else {
+      res += c;
+    }
+  }
+  return res;
```

Comment:
Shorter and more efficient:
```suggestion
  std::string res = txt;
  for(auto &c : res) {
    if(c == '.') {
      c = "&#46;";
    } 
  }
  return res;
```

Reply:
c is a "char" here so this doesn't work.

Reply:
oh, that is irritating.